query-engine/connector-test-kit-rs/query-engine-tests/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+// Rules for writing tests:
+// - mod name + test name have to be unique in name across all test suites.
+// - tests must be annotated with `connector_test`
+// - test modules can be annotated with `test_suite`. you get some niceties like imports and the ability to define
+// - you can use ONE OF `only` or `exclude` to scope connectors.
+//    - if you use none, the test is valid for all connectors.
+//
+// Notes:
+// - Allow dead code should be set?
+
+use query_test_macros::{connector_test, test_suite};
+
+pub mod schemas {
+    // Wild idea: validate schemas at compile time
+    pub fn some_common_schema() -> String {
+        "model C {
+            id Int @id
+            field String?
+        }"
+        .to_owned()
+    }
+}
+
+// The mod name dictates the db name. If the name is `some_spec`
+// then, for example, the MySQL db should be (similar to) `some_spec` as well.
+// #[cfg(test)]
+// #[before_each(before_each_handler)] // Hook to run before each test.
+// #[schema(schema_handler)] // Schema for all contained tests. Allows us to cache runners maybe.
+#[test_suite(schema(schemas::some_common_schema), only(Postgres(9)))]
+mod some_spec {
+
+    // These imports are required if no `#[test_suite(...)]` is used
```

Comment:
You should add the imports to the macro instead. Better to have them there than in ever test file...